pytests/validation/test_cross_valildation.py

```diff
@@ -115,29 +73,46 @@ def test_create_folds_splits(self, iterator, cv):
+                                dashify_logging_path: str,
+                                keep_interim_results: bool):
+        cross_validator = ValidatorFactory.get_cross_validator(gs_config=gs_cv_config,
+                                                               cv_config=cv_config,
+                                                               grid_search_id=grid_search_id,
+                                                               blue_print_type=blue_print_type,
+                                                               re_eval=False,
+                                                               keep_interim_results=keep_interim_results)
+        blueprints = cross_validator.create_blue_prints(blue_print_type=blue_print_type,
+                                                        gs_config=gs_cv_config,
+                                                        dashify_logging_path=dashify_logging_path,
+                                                        num_epochs=num_epochs,
+                                                        job_type=job_type)
+        gs_config = GridSearch.create_gs_from_config_dict(gs_cv_config)
+        assert len(blueprints) == len(gs_config) * len(cross_validator._get_fold_indices())
+        for i, blueprint in enumerate(blueprints):
+            blueprint.config['cv_experiment_information']["config"]['experiment_id'] = i
+            blueprint.config['cv_experiment_information']["config"]['val_fold_id'] = i
+            blueprint.job_type = AbstractGymJob.Type.STANDARD
+            blueprint.run_id = str(i)
+            blueprint.epochs = num_epochs
+
+    @pytest.mark.parametrize("job_type", [AbstractGymJob.Type.STANDARD])
+    def test_run(self, blue_print_type: Type[BluePrint], job_type: AbstractGymJob.Type,
+                 gs_cv_config: Dict[str, Any], cv_config: Dict[str, Any], grid_search_id: str, num_epochs: int,
+                 dashify_logging_path: str, process_count: int, device_ids: List[int], log_std_to_file: bool,
+                 log_dir_path: str, keep_interim_results: bool, start_logging):
+        iterator_key = cv_config["CV"]["iterator_key"]
+        split_key = cv_config["CV"]["split_key"]
+        component_names = [iterator_key]
+        components = blue_print_type.construct_components(config=gs_cv_config, component_names=component_names)
+        iterator = components[iterator_key][split_key]
+        cross_validator = CrossValidation(dataset_iterator=iterator, grid_search_id=grid_search_id,
+                                          **cv_config["CV"]["config"],
+                                          re_eval=False,
+                                          keep_interim_results=keep_interim_results)
+        gym = Gym(process_count, device_ids=device_ids, log_std_to_file=log_std_to_file)
+        cross_validator.run(blue_print_type, gym, gs_cv_config, num_epochs, dashify_logging_path)
+        QueuedLogging.stop_listener()
```

Comment:
do you think, we can also have asserts here? Maybe we can read in the log files and e.g.,check the metrics values? I know we don't have a proper training dataset, but since it's seeded we could check the values no?

Reply:
done,